lib/spack/spack/parser.py

```diff
@@ -96,7 +96,7 @@
 QUOTED_VALUE = r"[\"']+([a-zA-Z_0-9\-+\*.,:=\~\/\\\s]+)[\"']+"
 
 VERSION = r"=?([a-zA-Z0-9_][a-zA-Z_0-9\-\.]*\b)"
-VERSION_RANGE = rf"({VERSION}\s*:\s*{VERSION}(?!\s*=)|:\s*{VERSION}(?!\s*=)|{VERSION}\s*:|:)"
+VERSION_RANGE = rf"({VERSION}:{VERSION}(?!=)|:{VERSION}(?!=)|{VERSION}:|:)"
```

Comment:
```suggestion
VERSION_RANGE = rf"({VERSION}:{VERSION}|:{VERSION}|{VERSION}:|:)"
```

I guess negative lookahead can now also go.

Under the current rules `@:abc=def` parses as `@: abc=def`, but maybe better to parse a a range `@:abc` (and then a syntax error?)